### ISSUE TITLE
package: doit/ eli5

### ISSUE THREAD

**Issue by InLaw** [cript0nauta/pynixify#44]
```
Executing pipInstallPhase
/build/eli5-0.10.1/dist /build/eli5-0.10.1
Processing ./flufl.enum-4.1.1-py3-none-any.whl
Processing ./doit-0.32.0-py3-none-any.whl
Processing ./geolinks-0.2.0-py3-none-any.whl
ERROR: Could not find a version that satisfies the requirement pyinotify; sys_platform == "linux" (from doit==0.32.0) (from versions: none)
ERROR: No matching distribution found for pyinotify; sys_platform == "linux" (from doit==0.32.0)
Installing collected packages: flufl.enum
Installing collected packages: geolinks
Successfully installed flufl.enum-4.1.1
Successfully installed geolinks-0.2.0
builder for '/nix/store/f5m2r33m0794pbkcimjan3l0yp3x5jly-python3.7-doit-0.32.0.drv' failed with exit code 1
cannot build derivation '/nix/store/0yg3aw51l3xmfpl4616kr9axaypavl4i-python3-3.7.8-env.drv': 1 dependencies couldn't be built
error: --- Error --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix-shell
build of '/nix/store/0yg3aw51l3xmfpl4616kr9axaypavl4i-python3-3.7.8-env.drv' failed


```

**Comment by cript0nauta** [comment]
The case with doit is quite tricky. In nixpkgs, it is defined as a general application and not as a reusable Python package: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/build-managers/doit/default.nix#L8. This makes pynixify use the PyPI version. Building it fails because doit doesn't depend on pynixify using `setup_requires`. It uses `extras_requires` in a strange way: https://github.com/pydoit/doit/blob/master/setup.py#L83.

The [setuptools docs on platform specific dependencies](https://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-platform-specific-dependencies) doesn't say anything about using `extras_require` to indicate them, so I'm not sure how to proceed with this.

**Comment by cript0nauta** [comment]
Fixed in 4f0c35e453324d23ee70357baea052538f0f872c.
